### DISCUSSION TITLE
YOLOV8 train support Ray cluster?

### DISCUSSION THREAD

**Discussion by tangjn** [ultralytics/ultralytics#16469] [Q&A] [answered]
Hi Guys, 

I could find the ray tune from the document https://docs.ultralytics.com/integrations/ray-tune/  
For the Yolov8 train, I also want to use ray cluster. How do I use it or is any document?  Thanks.
eg:  I would like to use:    model.train(.....,   use_ray=True).

Thanks.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tangjn, thank you for your interest in Ultralytics 🚀! It sounds like you're exploring integrating Ray clusters with YOLOv8 for training. While our [Ray Tune integration docs](https://docs.ultralytics.com/integrations/ray-tune/) cover tuning aspects, using Ray clusters for training might require additional configurations.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If you have a custom ❓ Question or need help with configurations, please provide detailed information such as your current setup, any error messages, and logs. This will assist our engineers in guiding you more effectively.

For more interactive assistance, you can connect with our community in several ways:
- 🎧 Join our [Discord](https://ultralytics.com/discord) for real-time chat 
- Engage in deeper discussions on [Discourse](https://community.ultralytics.com)
- Or, explore threads on our [Subreddit](https://reddit.com/r/ultralytics) 

## Upgrade

Ensure your `ultralytics` package and dependencies are up-to-date to leverage all features:

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run in various environments, so choose the one that best suits your needs:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Services** with dedicated guides for [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) and [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Follow our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing.

This is an automated response, but rest assured an Ultralytics engineer will follow up to provide additional assistance soon! 😊

**Comment by glenn-jocher** [comment, marked as answer]
@tangjn hello,

Currently, YOLOv8 supports Ray Tune for hyperparameter tuning, but not direct training with Ray clusters. For distributed training, consider using other distributed computing frameworks. Let us know if you have further questions!
